src/partial_quicksync.rs

```diff
@@ -160,15 +160,30 @@ pub fn partial_restore(
   let client = Client::new();
   let conn = Connection::open(target_db_path)?;
   let user_version = get_user_version(&conn)?;
-  let remote_metadata = client
+  let response = client
     .get(format!(
       "{}/{}/metadata.csv?version={}",
       base_url,
       user_version,
       env!("CARGO_PKG_VERSION")
     ))
-    .send()?
-    .text()?;
+    .send()
+    .context(format!(
```

Comment:
Nit: with_context is lazy:
```suggestion
    .with_context(|| format!(
```